### PR TITLE
Remove Patroni leftover objects on cluster deletion.

### PR DESCRIPTION
**Pull request by alexeyklyukin**
To address #241

### CODE REVIEW

**Comment by erthalion** [comment]
:+1:

**Comment by Jan-M** [comment]
:+1:

**Comment by alexeyklyukin** [comment]
👍

File: pkg/cluster/cluster.go [hunk 1]
```diff
@@ -817,3 +822,57 @@ func (c *Cluster) shouldDeleteSecret(secret *v1.Secret) (delete bool, userName s
...
+	}
+	return nil
+}
+
+// TODO: figure out the interface shenanigans to avoid code duplication below.
+func (c *Cluster) deletePatroniClusterEndpoints() error {
+	for _, suffix := range patroniObjectSuffixes {
+		configEndpointName := fmt.Sprintf("%s-%s", c.Name, suffix)
+
+		if ep, err := c.KubeClient.Endpoints(c.Namespace).Get(configEndpointName, metav1.GetOptions{}); err == nil {
```
Comment (erthalion) [inline pkg/cluster/cluster.go]: Why not pass `c.KubeClient.Endpoints(c.Namespace)` as an argument here? In this way we can avoid this code duplication.

Comment (alexeyklyukin) [reply pkg/cluster/cluster.go]: I don't think it is that easy. Endpoint and ConfigMap implement 2 different interfaces and Get functions actually return different types (*v1.Endpoint and *v1.ConfigMap and respectively).

**Comment by alexeyklyukin** [comment]
👍

**Comment by erthalion** [comment]
:+1:

**Comment by alexeyklyukin** [comment]
👍

**Pull request merged**